src/posit/connect/external/databricks.py

```diff
@@ -160,277 +91,401 @@ def __call__(self) -> Dict[str, str]:
+    `databricks_config()` helper method in this module.
 
-    Examples
+    See Also
     --------
-    In the example below, the `PositContentCredentialsStrategy` can be initialized anywhere that
-    the Python process can read environment variables.
+    * https://docs.posit.co/ide/server-pro/user/posit-workbench/guide/databricks.html#databricks-with-python
 
-    CLIENT_ID and CLIENT_SECRET are credentials associated with the Databricks service principal.
+    Examples
+    --------
+    This example shows how authenticate to Databricks using Posit Workbench-managed Databricks Credentials.
 
     ```python
-    from posit.connect.external.databricks import (
-        PositContentCredentialsStrategy,
-        PositLocalContentCredentialsStrategy,
-    )
+    import os
 
-    import pandas as pd
-    from databricks import sql
     from databricks.sdk.core import ApiClient, Config
     from databricks.sdk.service.iam import CurrentUserAPI
+    from shiny import reactive
+    from shiny.express import render
 
-    DATABRICKS_HOST = "<REDACTED>"
-    DATABRICKS_HOST_URL = f"https://{DATABRICKS_HOST}"
-    SQL_HTTP_PATH = "<REDACTED>"
-    TOKEN_ENDPOINT_URL = f"https://{DATABRICKS_HOST}/oidc/v1/token"
-
-    CLIENT_ID = "<REDACTED>"
-    CLIENT_SECRET = "<REDACTED>"
-
-    # Rather than relying on the Databricks CLI as a local strategy, we use
-    # PositLocalContentCredentialsStrategy as a drop-in replacement.
-    # Can be replaced with the Databricks CLI implementation when
-    # https://github.com/databricks/cli/issues/1939 is resolved.
-    local_strategy = PositLocalContentCredentialsStrategy(
-        token_endpoint_url=TOKEN_ENDPOINT_URL,
-        client_id=CLIENT_ID,
-        client_secret=CLIENT_SECRET,
-    )
+    from posit.connect.external.databricks import WorkbenchStrategy
 
-    posit_strategy = PositContentCredentialsStrategy(local_strategy=local_strategy)
 
-    cfg = Config(host=DATABRICKS_HOST_URL, credentials_strategy=posit_strategy)
+    @reactive.calc
+    def cfg():
+        return Config(
+            credentials_strategy=WorkbenchStrategy(),
+            host=os.getenv("DATABRICKS_HOST"),
+        )
 
-    databricks_user_info = CurrentUserAPI(ApiClient(cfg)).me()
-    print(f"Hello, {databricks_user_info.display_name}!")
 
-    query = "SELECT * FROM samples.nyctaxi.trips LIMIT 10;"
-    with sql.connect(
-        server_hostname=DATABRICKS_HOST,
-        http_path=SQL_HTTP_PATH,
-        credentials_provider=posit_strategy.sql_credentials_provider(cfg),
-    ) as connection:
-        with connection.cursor() as cursor:
-            cursor.execute(query)
-            rows = cursor.fetchall()
-            print(pd.DataFrame([row.asDict() for row in rows]))
+    @render.text
+    def text():
+        databricks_user_info = CurrentUserAPI(ApiClient(cfg())).me()
+        return f"Hello, {databricks_user_info.display_name}!"
```

Comment:
nit; extract variables to improve readability.

```suggestion
        current_user_api = CurrentUserAPI(ApiClient(cfg()))
        databricks_user_info = current_user_api.me()
        return f"Hello, {databricks_user_info.display_name}!"
```